src/dune_engine/build_config.ml

```diff
@@ -110,11 +110,6 @@ let set
         contexts
         ~f:(fun ((ctx : Build_context.t), ctx_type) -> ctx.name, (ctx, ctx_type)))
   in
-  let () =
-    match (cache_config : Dune_cache.Config.t) with
```

Comment:
@snowleopard. Is moving this to the caller acceptable? Seems like this is an improvement to allow the caller to create the cache directories and handle any possible errors.